functions/snippets/.gcloudignore

```diff
@@ -0,0 +1 @@
+src/test/
```

Comment:
Is this file auto-generate?

Reply:
Could you expand on why you've chosen to do include this?

My understanding is that the Functions deployment deploys a jar? Are they compiled locally or after the code is deployed? 

If it's compiled locally, I'm not sure this has any effect, and it's done remotely I'm not sure if it's a pattern we should push. As a user, if I've written tests, I want those tests to run on compile.

Reply:
With GCF, you can deploy either a (self-sufficient) jar or a source hierarchy with `pom.xml` at the root. We expect users to deploy from source; the jar business is there to support build systems other than Maven. If you include the test source then your tests will be run remotely as part of your deployment. Whether you want that probably depends whether running them locally is enough, and on how long they take to run.

I think we should probably talk about this more in the GCF Java docs. We could also have a comment here saying what this does and why you might or might not want it. `#` comments are supported in `.gcloudignore`.

Reply:
I think that's the behavior most people will want though right? There might be some folks that don't want to run their tests on deploy, but I don't think it should be the default action.

From my perspective as a user, I would prefer to run `gcloud functions deploy` and have it run the tests + deploy rather than have to run `mvn verify` then run `gcloud functions deploy`.